features/step_definitions/machine.rb

```diff
@@ -71,8 +71,10 @@
     end
     
     machines_waiting_delete.each do | machine |
+      step %Q/I successfully merge patch resource "machine\/#{machine.name}" with:/, table(%{
+         | {"metadata":{"finalizers":[]}} |
```

Comment:
I think we need a more thorough discussion before merging this. For any machine waiting delete, we should let the controller delete it itself. If we remove the finalizer, can its node still be successfully deleted before deleting the machine? We can implement this if we know node is always deleted before machine is deleted, without the finalizers.

Reply:
And I propose that removing the finalizer should be handler as an exception. For example: given a certain time has passed, say 900 seconds, machine can not ensure_deleted, then we remove it's finalizer.

Reply:
Makes sense. Do we have understanding why removing with finalizers fails?